docs/upgrade.adoc

```diff
@@ -5,6 +5,62 @@
 Please follow these instructions when upgrading from Titan or an older
 JanusGraph release.
 
+=== Upgrading to JanusGraph 0.3.0
+
+JanusGraph 0.3.0 implements <<schema-constraints, schema constraints>> which made it necessary to also introduce the concept of a schema version. There is a check to prevent client connections that either expect a different schema version or have no concept of a schema version. To perform an upgrade, the configuration option `graph.allow-upgrade=true` must be set on each graph you wish to upgrade. The graph must be opened with a 0.3.0 or greater version of JanusGraph since older versions have no concept of `graph.storage-version` and will not allow for it to be set.
+
+Example excerpt from `janusgraph.properties` file
+[source, text]
+----
+# JanusGraph configuration sample: Cassandra over a socket
+#
+# This file connects to a Cassandra daemon running on localhost via
+# Thrift.  Cassandra must already be started before starting JanusGraph
+# with this file.
+
+# This option should be removed as soon as the upgrade is complete. Otherwise in the future
+# if this file was used to connect to a different graph it could cause and unintended upgrade.
+graph.allow-upgrade=true
+
+gremlin.graph=org.janusgraph.core.JanusGraphFactory
+
+# The primary persistence provider used by JanusGraph.  This is required.
+# It should be set one of JanusGraph's built-in shorthand names for its
+# standard storage backends (shorthands: berkeleyje, cassandrathrift,
+# cassandra, astyanax, embeddedcassandra, cql, hbase, inmemory) or to the
+# full package and classname of a custom/third-party StoreManager
+# implementation.
+#
+# Default:    (no default value)
+# Data Type:  String
+# Mutability: LOCAL
+storage.backend=cassandrathrift
+
+# The hostname or comma-separated list of hostnames of storage backend
+# servers.  This is only applicable to some storage backends, such as
+# cassandra and hbase.
+#
+# Default:    127.0.0.1
+# Data Type:  class java.lang.String[]
+# Mutability: LOCAL
+storage.hostname=127.0.0.1
+
+----
+
+*IMPORTANT* You should back-up your data prior to attempting an upgrade!
```

Comment:
A migration from Titan, however, is unable to detect that the storage format has changed since Titan would not know how to determine this (`titan.version` and `titan.compatible-versions` are essentially hardcoded in the Titan runtime).

Please add text to the [Migrating from Titan](https://github.com/JanusGraph/janusgraph/blob/master/docs/migrating.adoc) documentation to warn them that once they have migrated to JanusGraph 0.3.0 that they should not connect to the graph instance again with Titan.

Reply:
I was thinking I could also add the line below to the upgrade to prevent titan connections
```
globalWrite.set(TITAN_COMPATIBLE_VERSIONS, JanusGraphConstants.VERSION);
```

TITAN_COMPATIBLE_VERSIONS sets graph.titan-version which lines up with INITIAL_TITAN_VERSION. Titan has a check on graph.titan-version that should prevent connection.